backends/pkg/reconciler/eventmesh/handler.go

```diff
@@ -234,39 +237,41 @@ func collectSubscribedEventTypes(trigger *eventingv1.Trigger, broker *Broker, et
 }
 
 // fetchBrokers fetches the brokers and converts them to the representation that's consumed by the Backstage plugin.
-func fetchBrokers(brokerLister eventinglistersv1.BrokerLister, logger *zap.SugaredLogger) ([]*Broker, error) {
-	fetchedBrokers, err := brokerLister.List(labels.Everything())
+func fetchBrokers(clientset versioned.Interface, logger *zap.SugaredLogger) ([]*Broker, error) {
+	brokers, err := clientset.EventingV1().Brokers("default").List(context.Background(), metav1.ListOptions{})
+
 	if err != nil {
 		logger.Errorw("Error listing brokers", "error", err)
 		return nil, err
 	}
 
-	convertedBrokers := make([]*Broker, 0, len(fetchedBrokers))
-	for _, br := range fetchedBrokers {
-		convertedBroker := convertBroker(br)
+	convertedBrokers := make([]*Broker, 0, len(brokers.Items))
+	for _, br := range brokers.Items {
+		convertedBroker := convertBroker(&br)
 		convertedBrokers = append(convertedBrokers, &convertedBroker)
 	}
 	return convertedBrokers, err
 }
 
 // fetchEventTypes fetches the event types and converts them to the representation that's consumed by the Backstage plugin.
-func fetchEventTypes(eventTypeLister eventinglistersv1beta2.EventTypeLister, logger *zap.SugaredLogger) ([]*EventType, error) {
-	fetchedEventTypes, err := eventTypeLister.List(labels.Everything())
+func fetchEventTypes(clientset versioned.Interface, logger *zap.SugaredLogger) ([]*EventType, error) {
+	eventTypeResponse, err := clientset.EventingV1beta2().EventTypes("default").List(context.Background(), metav1.ListOptions{})
```

Comment:
Can we try fetching all event types from all namespaces like before?

I think the reason you converted this code to check out a specific namespace is the token permissions.

We will instruct the users of the plugin to set up a token for a ServiceAccount that can fetch all eventTypes+triggers+etc. in all namespaces.

Reply:
Same story for brokers and triggers too

Reply:
To create an SA and a token, you can do this:

```bash
# 1. create a new service account in the kube-system namespace
kubectl -n kube-system create serviceaccount backstage-admin
# 2. create a clusterrolebinding
kubectl create clusterrolebinding backstage-admin --clusterrole=cluster-admin --serviceaccount=kube-system:backstage-admin
# 3. create a secret for the service account
kubectl apply -f - <<EOF
    apiVersion: v1
    kind: Secret
    metadata:
      name: backstage-admin
      namespace: kube-system
      annotations:
        kubernetes.io/service-account.name: backstage-admin
    type: kubernetes.io/service-account-token
EOF
```

This will obviously give cluster-admin superpowers to the ServiceAccount and thus the token, but it will make sure there's access to the eventTypes/brokers/triggers.


To get a token for this SA:

```bash
export KUBE_SA_TOKEN=$(kubectl -n kube-system get secret backstage-admin -o jsonpath='{.data.token}' | base64 --decode)
```

You can also run a sanity check with the token:

```bash
export KUBE_API_SERVER_URL=$(kubectl config view --minify --output jsonpath="{.clusters[*].cluster.server}") # e.g. "https://192.168.2.151:16443"

# run a sanity check with the token
curl -k -H "Authorization: Bearer $KUBE_SA_TOKEN" -X GET "${KUBE_API_SERVER_URL}/api/v1/nodes" | json_pp
```



We'll eventually remove the cluster role here: https://github.com/knative-extensions/backstage-plugins/blob/main/backends/config/100-eventmesh/100-clusterrole.yaml

and the rolebinding for it.

That will mean we won't specify what permissions to use in the deployment and only ask for a token in the HTTP request.